docs/FormatsAndConventions.md

```diff
@@ -138,6 +138,25 @@ e.g. `Error alert email has very long subject #5958`
 > to compensate for GitHub's lack of strong linking between the two.
 > Assume there is an invisible prefix in front of the PR title `Fixes issue : ...`
 
+## References to code elements
+
+Follow these conventions when referring to code elements e.g. when referring to a function name from a commit message.
+The objective is to be as concise as possible without causing ambiguities. Therefore, omit optional details when
+those details are not pertinent to the context.
+
+### Java
+
+* Variables: `package.class#variable` (optional: `package`)
+* Methods: `package.class#method(paramTypes):returnType` (optional: `package`, `paramTypes`, `:returnType`)
+
+e.g.
+* `Person#name`
+* `Person#getName()` 
+* `Person#getName():String` 
+* `Person#setName()` -- parameters omitted because only one `setName(String)` in the `Person` class
```

Comment:
How about this: we don't make `paramType` optional but adds the following escape clause: 

> In cases where specifying `paramTypes` pushes against a severe length restriction (e.g. commit message title), you may replace `paramTypes` with a `...`.

Note that it is the commit message title that is problematic. The body can be as long as we want although 72 char limit makes it awkward to have long method signatures in it.

Use of ellipsis adds 3 extra chars but avoids confusion between 'omitting parameters' and 'no parameters'

Also, I'm fine with specifying the full signature once and using the shortened version (i.e. with `...`) in the subsequent references.